trustymail/domain.py

```diff
@@ -142,7 +143,8 @@ def generate_results(self):
             "DMARC Results on Base Domain": self.parent_dmarc_results(),
             "DMARC Policy": self.get_dmarc_policy(),
             
-            "Syntax Errors": self.format_list(self.syntax_errors)
+            "Syntax Errors": self.format_list(self.syntax_errors),
+            "Errors": self.format_list(self.errors)
```

Comment:
Because we embed scan results in our reports, and the attachments do get (occasionally 😅) looked at, the more clarity we can provide up front to enable self-servicing the better, lest we get sent a ton of inquiries. I agree with #27 that this should probably be renamed, and clarified. For example, the README indicates some of this ambiguity. A snippit reads:

> * Syntax Errors - A list of syntax errors that were detected when scanning DMARC or SPF records, or checking for STARTTLS support.
> * Errors - A list of any other errors encountered, such as DNS failures.

If `Syntax Errors` are just for errors of syntax, network error information about STARTTLS should probably not go there.

Perhaps `Debug` is a better field name than `Errors`? 

Also, the information that lands in `Errors` can be ambiguous as to its provenance. For instance, in a scan of one domain, the error field results were:

```bash
[Errno 101] Network is unreachable, [Errno 101] Network is unreachable, [Errno 101] Network is unreachable, [Errno 101] Network is unreachable, [Errno 101] Network is unreachable
```
There actually was a network-related error. However, the error output is not clear as to the object whose network appears unreachable. (The domain scanned has 5 mail servers listed on their MX record, so I expect this error is in reference to those 5 hostnames.)